plugins/modules/ec2_instance.py

```diff
@@ -54,7 +54,6 @@
     description:
       - Instance type to use for the instance, see U(https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html)
         Only required when instance is not already present.
-    default: t2.micro
```

Comment:
Removing the actual module default value is probably our best option, but we should add a note here that if no option is specified, t2.micro will be used (so it isn't a surprise if someone just forgets to specify or something).  We should also add a deprecation warning to the code about that the default value is going to go away and in the future folks will always need to specify either an instance_type, or an LT.  Then we can actually deprecate it and avoid this entirely in the future!